src/skillsaw/rules/builtin/mcp_registry/server_json_valid.py

```
@@ -109,6 +204,93 @@ def _mapping_item(data: dict, key: str, index: object) -> Optional[dict]:
+    package: dict,
+    schema_profile: McpRegistrySchemaProfile,
+) -> frozenset[str]:
+    """Collect URL variables from the package fields the publisher resolves."""
+    variables = set()
+    environment = package.get(schema_profile.environment_variables_field)
+    if isinstance(environment, list):
+        for item in environment:
+            name = item.get("name") if isinstance(item, dict) else None
+            if isinstance(name, str) and name:
+                variables.add(name)
+    for field in (
```

Ruff (0.16.2)

[error] Import `field` from line 6 shadowed by loop variable

(F402)